mpqp/noise/noise_model.py

```diff
@@ -277,25 +277,103 @@ def to_other_language(
 
 
 class BitFlip(NoiseModel):
-    """3M-TODO"""
+    """Class representing the bit flip noise channel, which flips the state of
+    a qubit with a certain probability. It can be applied to single-qubit gates
+    and depends on a single parameter (probability or error rate).
 
-    # def __init__(
-    #     self,
-    #     proba: Union[float, Expr],
-    #     targets: List[int],
-    #     dimension: int = 1,
-    #     gates: List[Gate] = None):
+    Args:
+        prob: Bit flip error probability or error rate (must be within [0, 0.5]).
+        targets: List of qubit indices affected by this noise.
+        gates: List of :class:`Gates<mpqp.core.instruction.gates.gate.Gate>`
+            affected by this noise. Only single-qubit gates are valid.
 
-    #     super().__init__(proba, targets, dimension, gates)
+    Raises:
+        ValueError: When the probability is outside of the expected interval [0, 0.5].
+        ValueError: When no target qubits are specified.
+        ValueError: When the specified gates are not single-qubit gates.
 
-    def to_kraus_representation(self) -> KrausRepresentation:
-        # generate Kraus operators for bit flip noise
-        # kraus_operators = [
-        #     np.sqrt(1 - self.proba) * np.array([[1, 0], [0, 1]]),  # Identity
-        #     np.sqrt(self.proba) * np.array([[0, 1], [1, 0]])      # Bit flip
-        # ]
-        # return KrausRepresentation(kraus_operators)
-        ...
+    Examples:
+        >>> circuit = QCircuit([H(i) for i in range(3)])
+        >>> bf1 = BitFlip(0.1, [0])
+        >>> bf2 = BitFlip(0.3, [1, 2])
+        >>> bf3 = BitFlip(0.05, [0], gates=[H])
+        >>> circuit.add([bf1, bf2, bf3])
+        >>> print(circuit)
+             ┌───┐
+        q_0: ┤ H ├
+             ├───┤
+        q_1: ┤ H ├
+             ├───┤
+        q_2: ┤ H ├
+             └───┘
+        NoiseModel:
+            BitFlip(prob=0.1, targets=[0])
+            BitFlip(prob=0.3, targets=[1, 2])
+            BitFlip(prob=0.05, targets=[0], gates=[H])
+
+    """
+
+    def __init__(
+        self,
+        prob: float,
+        targets: Optional[list[int]] = None,
+        gates: Optional[list[type[Gate]]] = None,
+    ):
+
+        if not (0 <= prob <= 0.5):
+            raise ValueError(
+                f"Invalid probability: {prob} but should be between 0 and 0.5"
+            )
+
+        if gates is not None:
+            if any(gate.nb_qubits != 1 for gate in gates):
```

Comment:
If we choose to extend bit-flip to multi qubit gates, then we should remove this

Reply:
Exactly, got removed